bundles/org.openhab.binding.easee/src/main/java/org/openhab/binding/easee/internal/model/GenericResponseTransformer.java

```diff
@@ -0,0 +1,126 @@
+                String channelType = channel.getAcceptedItemType();
+
+                if (value == null || channelType == null) {
+                    result.put(channel, UnDefType.NULL);
+                } else {
+                    try {
+                        switch (channelType) {
+                            case CHANNEL_TYPE_SWITCH:
+                                result.put(channel, OnOffType.from(Boolean.parseBoolean(value)));
+                                break;
+                            case CHANNEL_TYPE_VOLT:
+                                result.put(channel, new QuantityType<>(Double.parseDouble(value), Units.VOLT));
+                                break;
+                            case CHANNEL_TYPE_AMPERE:
+                                result.put(channel, new QuantityType<>(Double.parseDouble(value), Units.AMPERE));
+                                break;
+                            case CHANNEL_TYPE_KWH:
+                                result.put(channel, new QuantityType<>(Double.parseDouble(value),
+                                        MetricPrefix.KILO(Units.WATT_HOUR)));
+                                break;
+                            case CHANNEL_TYPE_KW:
+                                result.put(channel,
+                                        new QuantityType<>(Double.parseDouble(value), MetricPrefix.KILO(Units.WATT)));
+                                break;
+                            case CHANNEL_TYPE_DATE:
+                                result.put(channel, new DateTimeType(Utils.parseDate(value)));
+                                break;
+                            case CHANNEL_TYPE_STRING:
+                                result.put(channel, new StringType(value));
+                                break;
+                            case CHANNEL_TYPE_NUMBER:
+                                if (Utils.getChannelTypeId(channel).contains(CHANNEL_TYPENAME_INTEGER)) {
+                                    // explicit type long is needed in case of integer/long values otherwise automatic
+                                    // transformation to a decimal type is applied.
+                                    result.put(channel, new DecimalType(Long.parseLong(value)));
+                                } else {
+                                    result.put(channel, new DecimalType(Double.parseDouble(value)));
+                                }
+                                break;
+                            default:
+                                logger.warn("no mapping implemented for channel type '{}'", channelType);
+                        }
+
+                        // call the custom handler to handle specific / composite channels which do not map 1:1 to JSON
+                        // fields.
+                        result.putAll(customResponseTransformer.transform(channel, value, jsonData));
+
+                    } catch (Exception ex) {
```

Comment:
This class converts data retrieved via the API. It could happen that there is unexpected data retrieved. If I do not catch all excpetions (actually there are no checked exceptios at all) this will either stop the whole binding or at least stop updating other channels which are retrieved via the same API call.

Reply:
I made the catch block more specific. The exceptions are RuntimeExceptions but it makes sense to catch them.